admin.go

```diff
@@ -1043,72 +1064,111 @@ func (ca *clusterAdmin) ListConsumerGroups() (allGroups map[string]string, err e
 }
 
 func (ca *clusterAdmin) ListConsumerGroupOffsets(group string, topicPartitions map[string][]int32) (*OffsetFetchResponse, error) {
-	coordinator, err := ca.client.Coordinator(group)
-	if err != nil {
-		return nil, err
-	}
-
+	var response *OffsetFetchResponse
 	request := NewOffsetFetchRequest(ca.conf.Version, group, topicPartitions)
+	err := ca.retryOnError(isRetriableGroupCoordinatorError, func() error {
+		coordinator, err := ca.client.Coordinator(group)
+		if err != nil {
+			return err
+		}
 
-	return coordinator.FetchOffset(request)
+		defer func() {
+			if err != nil && isRetriableGroupCoordinatorError(err) {
+				_ = ca.client.RefreshCoordinator(group)
+			}
+		}()
```

Comment:
If you’re going to be accessing the `err` in a `defer` like this, it’s recommended to then name the return up on line 1069: `func() (err error)` this means that the properly scoped `err` will be set by the `return ErrWhatever`

This then prevents accidentally missing an `err = ErrWhatever` and even more so, guards against an (accidental) shadowing of `err` from preventing the ability to be able to do that assignment at all.

Reply:
Yeah I had planned to make that change, was still just getting something together that works in practice and passes the retry tests 😅

Reply:
👍 Work first, polish second. :smart: